src/tests/multihost/ipa/conftest.py

```diff
@@ -165,6 +166,83 @@ def restoresssdconf():
+    client.transport.put_file(os.getcwd() +
+                              '/src/tests/multihost/ipa/data/list_subid_ranges.c',
+                              '/tmp/list_subid_ranges.c')
+    client.run_command("gcc /tmp/list_subid_ranges.c"
+                       "  -lsubid -o /tmp/list_subid_ranges")
+
+    def remove():
+        """ Remove file """
+        for file in ['list_subid_ranges', 'list_subid_ranges.c']:
+            client.run_command(f"rm -vf /tmp/{file}")
+
+    request.addfinalizer(remove)
+
+
+@pytest.fixture(scope='class')
+def subid_generate(session_multihost, request):
+    """
+    Generate subid for user admin
+    """
+    user = "admin"
+    test_password = "Secret123"
+    ssh1 = SSHClient(session_multihost.client[0].ip,
+                     username=user, password=test_password)
+    (result, result1, exit_status) = ssh1.execute_cmd('kinit',
+                                                      stdin=test_password)
+    assert exit_status == 0
+    (result, result1, exit_status) = ssh1.exec_command('klist')
+    assert user in str(result1.read())
+    (result, result1, exit_status) = ssh1.exec_command(f"ipa "
+                                                       f" subid-generate"
+                                                       f"  --owner={user}")
+    ssh1.close()
+
+
+@pytest.fixture(scope='class')
+def bkp_cnfig_for_subid_files(session_multihost, request):
+    """ Back up files used in test
+        And config /etc/nsswitch.conf
+    """
+    session_multihost.client[0].run_command("cp -vf  "
+                                            "/etc/subuid "
+                                            "/tmp/subuid_bkp")
+    session_multihost.client[0].run_command("cp -vf  "
+                                            "/etc/subgid "
+                                            "/tmp/subgid_bkp")
+    session_multihost.client[0].run_command("cp -vf  "
+                                            "/etc/nsswitch.conf "
+                                            "/tmp/nsswitch.conf_bkp")
+    session_multihost.client[0].run_command("echo 'subid: sss'  "
+                                            ">> /etc/nsswitch.conf")
+
+    def restore():
+        """ Restore """
+        session_multihost.client[0].run_command("mv -vf  "
+                                                "/tmp/subuid_bkp "
+                                                "/etc/subuid")
+        session_multihost.client[0].run_command("mv -vf  "
+                                                "/tmp/subgid_bkp "
+                                                "/etc/subgid")
+        session_multihost.client[0].run_command("mv -vf  "
+                                                "/tmp/nsswitch.conf_bkp "
+                                                "/etc/nsswitch.conf")
+        session_multihost.client[0].run_command("rm -vf "
+                                                "/tmp/unshare.pid")
```

Comment:
This file is created only in one test, having this in class fixture is confusing.
I think it's better to remove this file in the test (if needed at all).

Reply:
fixed